charts/cluster-addons/templates/intel-device-plugin/monitoring.yaml

```diff
@@ -0,0 +1,109 @@
+{{- if and .Values.intelDevicePlugin.enabled .Values.intelDevicePlugin.xpuManagerMonitoring.enabled }}
+---
+apiVersion: addons.stackhpc.com/v1alpha1
+kind: Manifests
+metadata:
+  name: {{ include "cluster-addons.componentName" (list . "intel-xpumanager-services") }}
+spec:
+  clusterName: {{ include "cluster-addons.clusterName" . }}
+  targetNamespace: {{ .Values.intelDevicePlugin.xpuManagerMonitoring.namespace }}
+  releaseName: intel-xpumanager-services
+  manifestSources:
+  - template: |
+      #
+      # Copyright (C) 2021-2023 Intel Corporation
+      # SPDX-License-Identifier: MIT
+      # @file service-intel-xpum.yaml
+      #
+      apiVersion: v1
+      kind: Service
+      metadata:
+        labels:
+          app: intel-xpumanager
+        name: intel-xpumanager
+      spec:
+        ports:
+        - name: metrics
+          port: 29999
+          protocol: TCP
+        selector:
+          app: intel-xpumanager
+        sessionAffinity: None
+  - template: |
+      #
+      # Copyright (C) 2021-2023 Intel Corporation
+      # SPDX-License-Identifier: MIT
+      # @file servicemonitor-intel-xpum.yaml
+      #
+      apiVersion: monitoring.coreos.com/v1
+      kind: ServiceMonitor
+      metadata:
+        name: intel-xpumanager
+        labels:
+          app: intel-xpumanager
+      spec:
+        selector:
+          matchLabels:
+            app: intel-xpumanager
+        namespaceSelector:
+          matchNames:
+            - default
+        endpoints:
+          - port: metrics
+            path: /metrics
+            interval: 5s
+---
+apiVersion: source.toolkit.fluxcd.io/v1
+kind: GitRepository
+metadata:
+  name: intel-xpu-monitoring-repo
+spec:
+  interval: 5m
+  url: {{ .Values.intelDevicePlugin.xpuManagerMonitoring.repo.url }}
+  ref:
+    tag: {{ .Values.intelDevicePlugin.xpuManagerMonitoring.repo.tag }}
+---
+apiVersion: kustomize.toolkit.fluxcd.io/v1
+kind: Kustomization
+metadata:
+  name: intel-xpu-monitoring
```

Comment:
I think we should use cluster-addons.componentName here please, and also include the labels we have below.